zenoh/src/sample.rs

```diff
@@ -98,6 +98,193 @@ impl From<Option<DataInfo>> for SourceInfo {
     }
 }
 
+mod attachments {
+    #[zenoh_macros::unstable]
+    use zenoh_buffers::{reader::HasReader, writer::HasWriter, ZBuf, ZBufReader, ZSlice};
+    #[zenoh_macros::unstable]
+    use zenoh_codec::{LCodec, RCodec, WCodec, Zenoh080};
+    #[zenoh_macros::unstable]
+    use zenoh_protocol::zenoh::ext::AttachmentType;
+
+    #[zenoh_macros::unstable]
+    #[derive(Clone)]
+    pub struct Attachments {
+        pub(crate) inner: ZBuf,
+    }
+    #[zenoh_macros::unstable]
+    impl Default for Attachments {
+        fn default() -> Self {
+            Self::new()
+        }
+    }
+    #[zenoh_macros::unstable]
+    impl<const ID: u8> From<Attachments> for AttachmentType<ID> {
+        fn from(this: Attachments) -> Self {
+            AttachmentType { buffer: this.inner }
+        }
+    }
+    #[zenoh_macros::unstable]
+    impl<const ID: u8> From<AttachmentType<ID>> for Attachments {
+        fn from(this: AttachmentType<ID>) -> Self {
+            Attachments { inner: this.buffer }
+        }
+    }
+    #[zenoh_macros::unstable]
+    impl Attachments {
+        pub fn new() -> Self {
+            Self {
+                inner: vec![0u8].into(),
+            }
+        }
+        pub fn is_empty(&self) -> bool {
+            self.len() == 0
+        }
+        pub fn len(&self) -> usize {
+            Zenoh080.read(&mut self.inner.reader()).unwrap_or(0)
+        }
+        pub fn iter(&self) -> AttachmentsIterator {
+            self.into_iter()
+        }
+        fn _get(&self, key: &[u8]) -> Option<ZSlice> {
```

Comment:
Why we need `_get`, if it used only in the `get`? (the same for the insert)
Is it some rust syntax sugar?